src/Core/Form/IdentifiableObject/DataHandler/AttributeFormDataHandler.php

```diff
@@ -79,5 +81,16 @@ public function update($id, array $data)
             ->setAssociatedShopIds($data['shop_association']);
 
         $this->commandBus->handle($updateCommand);
+
+        // delete previous img if exist
+        if (file_exists(_PS_IMG_DIR_ . 'co/' . $id . '.jpg')) {
+            unlink(_PS_IMG_DIR_ . 'co/' . $id . '.jpg');
+        }
+        $this->handleUploadedFile($data['texture'], $id);
```

Comment:
This should be part of the command, you can inspire from what was done in other CQRS commands:
https://github.com/PrestaShop/PrestaShop/blob/fd1b5c16b7203bc418f31900a633c4370e9d5473/src/Core/Domain/Product/Image/Command/AddProductImageCommand.php
https://github.com/PrestaShop/PrestaShop/blob/3d06748975e00c548e940bf82862d9d211ae94d4/src/Core/Domain/Attachment/Command/AddAttachmentCommand.php

The commands should be passed with the path of the UploadFile (not an UploadFile instance, just a string), optionally you can also provide file size or mime type if needed

Reply:
Shouldn't the CQRS command reflect what is in the DB ? because the texture file path isn't in.  That's why I did not put it in the CQRS